src/libDirectoryService/DSBlockPreProcessing.cpp

```diff
@@ -241,7 +241,7 @@ void DirectoryService::RunConsensusOnDSBlock()
                       "To-do: Code up the logic for if we didn't get any "
                       "submissions at all");
             // throw exception();
-            return;
+            // return;
```

Comment:
What is the rationale of removing this return?

Reply:
Due to the synchronization interval might larger than the vacuous block duration, the moment when the DS fetched the latest states could be later than the PoW1 submission. In that case, this DS trying to rejoin will start the DSBlock consensus with no PoW1 submission, so it will continue the consensus by querying the missing PoW1 submission from other DS node. Commenting this return makes that happen.

Reply:
But what happen if, assuming without rejoining, the network received 0 submission of pows. It will skip pass the logic (not return) and may lead to a fatal crash.

Reply:
Added a bool for RunConsensuOnDSBlock if isRejoin with a default of false